File: mario.py
# Get the height input from the user and ensure it's between 1 and 8
while True:
    user_input = input("Height: ")

    # Check if the input is empty
    if user_input.strip() == "":
        print("Please enter a valid height (a number between 1 and 8).")
        continue

    # Try to convert the input to an integer
    try:
        height = int(user_input)

        # Check if the height is within the valid range
        if height < 1 or height > 8:
            print("Please enter a height between 1 and 8.")
        else:
            break  # Valid input, exit the loop
    except ValueError:
        # If conversion fails, it's not a number
        print("Please enter a valid number.")

# Now that we have a valid height, create the pyramid
for row in range(height):  # Loop through each row
    # Print the spaces
    for spaces in range(height - row - 1):
        print(" ", end="")  # Print spaces without a newline

    # Print the hashes
    for column in range(row + 1):
        print("#", end="")  # Print hashes without a newline

    # Print a newline after each row
    print()

File: mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{

    // prompt the user for a number for hight from 1 to 8, inclusive.

    int height;
    do
    {
        height = get_int("Height: ");
    }

    while (height < 1 || height > 8);

    // if the number meets the specifications create the pyramid

    // To make each row

    for (int row = 0; row < height; row++)
    {
        for (int spaces = 0; spaces < height - row - 1; spaces++)
        {
            printf(" ");
        }
        // To make each column
        for (int column = 0; column <= row; column++)
        {
            printf("#");
        }
        printf("\n");
    }
}
